### PR TITLE
[14.0][FIX] maintenance_plan: avoid error if planning_step is not defined

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Also, clean-up of useless from_string() calls.

### CODE REVIEW

**Review by mariadforgeflow — APPROVED** [review]
LGTM!

**Comment by MiquelRForgeFlow** [comment]
@etobella

File: maintenance_plan/models/maintenance_equipment.py [hunk 1]
```diff
@@ -67,13 +67,13 @@ def _prepare_request_from_plan(self, maintenance_plan, next_maintenance_date):
         return {
             "name": name,
             "request_date": next_maintenance_date,
-            "schedule_date": next_maintenance_date,
+            "schedule_date": fields.Datetime.to_datetime(next_maintenance_date),
```
Comment (etobella) [inline maintenance_plan/models/maintenance_equipment.py]: This on 14 is probably not necessary

File: maintenance_plan_activity/models/maintenance.py [hunk 2]
```diff
@@ -30,14 +30,14 @@ def _create_new_request(self, maintenance_plan):
...
-                            "Activity automatically generated from " "maintenance plan"
+                            "Activity automatically generated from maintenance plan"
                         ),
                         "user_id": planned_activity.user_id.id or self.env.user.id,
                         "res_id": request.id,
                         "res_model_id": self.env.ref(
                             "maintenance.model_maintenance_request"
                         ).id,
-                        "date_deadline": fields.Date.from_string(request.schedule_date)
+                        "date_deadline": fields.Date.to_date(request.schedule_date)
```
Comment (etobella) [inline maintenance_plan_activity/models/maintenance.py]: request.schedule_date should be enough

**Review by etobella — APPROVED** [review]

**Pull request merged**